### ISSUE TITLE
Cell references in conditional formatting expressions not updated when rows/columns are inserted/deleted

### ISSUE THREAD

**Issue by craigbrown** [EPPlusSoftware/EPPlus#476]
If you have a conditional formatting rule which is an expression, and the formula for the rule contains cell references, these cell references are not updated when rows/columns are inserted or deleted.

```csharp
[TestMethod]
public void CheckConditionalFormattingFormulaAfterDeletingColumn()
{
    using (var pck = new ExcelPackage())
    {
        // Create a worksheet with conditional formatting 
        var wks = pck.Workbook.Worksheets.Add("Sheet1");
        var cf = wks.ConditionalFormatting.AddExpression(wks.Cells["E2:E7"]);
        cf.Formula = "=($D2=0)";
        
        // Delete a column before the column being referenced by the CF formula
        wks.DeleteColumn(2);
        
        // Check the conditional formatting formula has been updated
        cf = (IExcelConditionalFormattingExpression)wks.ConditionalFormatting[0];
        Assert.AreEqual("=($C2=0)", cf.Formula);
    }
}

[TestMethod]
public void CheckConditionalFormattingFormulaAfterDeletingRow()
{
    using (var pck = new ExcelPackage())
    {
        // Create a worksheet with conditional formatting 
        var wks = pck.Workbook.Worksheets.Add("Sheet1");
        var cf = wks.ConditionalFormatting.AddExpression(wks.Cells["B5:G5"]);
        cf.Formula = "=(B$4=0)";
        
        // Delete a row before the column being referenced by the CF formula
        wks.DeleteRow(2);
        
        // Check the conditional formatting formula has been updated
        cf = (IExcelConditionalFormattingExpression)wks.ConditionalFormatting[0];
        Assert.AreEqual("=(B$3=0)", cf.Formula);
    }
}

[TestMethod]
public void CheckConditionalFormattingFormulaAfterInsertingColumn()
{
    using (var pck = new ExcelPackage())
    {
        // Create a worksheet with conditional formatting 
        var wks = pck.Workbook.Worksheets.Add("Sheet1");
        var cf = wks.ConditionalFormatting.AddExpression(wks.Cells["E2:E7"]);
        cf.Formula = "=($D2=0)";
        
        // Insert a column before the column being referenced by the CF formula
        wks.InsertColumn(2, 1);
        
        // Check the conditional formatting formula has been updated
        cf = (IExcelConditionalFormattingExpression)wks.ConditionalFormatting[0];
        Assert.AreEqual("=($E2=0)", cf.Formula);
    }
}

[TestMethod]
public void CheckConditionalFormattingFormulaAfterInsertingRow()
{
    using (var pck = new ExcelPackage())
    {
        // Create a worksheet with conditional formatting 
        var wks = pck.Workbook.Worksheets.Add("Sheet1");
        var cf = wks.ConditionalFormatting.AddExpression(wks.Cells["B5:G5"]);
        cf.Formula = "=(B$4=0)";
        
        // Insert a row before the column being referenced by the CF formula
        wks.InsertRow(2, 1);
        
        // Check the conditional formatting formula has been updated
        cf = (IExcelConditionalFormattingExpression)wks.ConditionalFormatting[0];
        Assert.AreEqual("=(B$5=0)", cf.Formula);
    }
}
```

**Comment by JanKallman** [comment]
Formulas for conditional formatting and data validations were not updated when performing delete and insert operation. I'll provide a  fix.

**Comment by JanKallman** [comment]
Fixed in version 5.7.5
